### PR TITLE
Warns

### PR DESCRIPTION
**Pull request by stsp**
Fixes gcc warnings.

### CODE REVIEW

**Comment by alexfru** [comment]
I'm already working on most of these.

File: v0100/smlrc.c [hunk 1]
```diff
@@ -1153,7 +1154,7 @@ void IncludeFile(int quot)
       }
       for (i = 0; i < pl; )
       {
-        int plen = strlen(paths + i);
+        size_t plen = strlen(paths + i);
```
Comment (alexfru) [inline v0100/smlrc.c:1157]: Which version of gcc, what options (just doing make?), what's the warning message?

File: v0100/smlrc.c [hunk 2]
```diff
@@ -3930,6 +3931,7 @@ int exprval(int* idx, int* ExprTypeSynPtr, int* ConstExpr)
         *ExprTypeSynPtr = -*ExprTypeSynPtr;
       else
         ++*ExprTypeSynPtr;
+      assert(*ExprTypeSynPtr >= 0);  // avoids gcc warning
```
Comment (alexfru) [inline v0100/smlrc.c:3934]: What's the warning message?

**Review by alexfru — COMMENTED** [review]
I'm not sure if or when I'm going to address these two in smlrc.c, but I'm curious about the details.

**Comment by stsp** [comment]
How come?
I put the error message in all
the log entries.

gcc is 14.0.1.
Yes, just running make.

**Comment by alexfru** [comment]
Sorry, I ignored the individual commits and their descriptions and just looked at the code changes.

**Pull request closed without merging**